### PR TITLE
Issue #2883: Add public TabCollectionStorage API to feature-tab-collections..

### PR DESCRIPTION
**Pull request by pocmo**
Once landed I'll add some more docs (and the changelog entry). Right now this is covered by "on-device" tests to actualyl invoke Room. But I would also like to add some unit tests to the mix to test more of the save/restore code properly.

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: components/feature/tab-collections/src/androidTest/java/mozilla/components/feature/tab/collections/TabCollectionStorageTest.kt [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+        getAllCollections().let { collections ->
+            assertEquals(1, collections.size)
+
+            assertEquals("Articles", collections[0].title)
+        }
+    }
+
+    private fun getAllCollections(): List<TabCollection> {
+        // Not sure exactly why; but without delaying here `getAllCollections` may not see the latest state.
```
Comment (Amejia481) [inline components/feature/tab-collections/src/androidTest/java/mozilla/components/feature/tab/collections/TabCollectionStorageTest.kt]: Are the user going to experience the same issue ?

File: components/feature/tab-collections/src/main/java/mozilla/components/feature/tab/collections/Tab.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+     */
+    val title: String
+
+    /**
+     * The URL of the tab.
+     */
+    val url: String
+
+    /**
+     * Restores a single tab from this collection and returns a matching returns a matching [SessionManager.Snapshot].
```
Comment (Amejia481) [inline components/feature/tab-collections/src/main/java/mozilla/components/feature/tab/collections/Tab.kt]: nit: **returns a matching** returns a matching

File: components/feature/tab-collections/src/main/java/mozilla/components/feature/tab/collections/adapter/TabAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        get() = actual.id!!
+
+    override val title: String
+        get() = actual.title
+
+    override val url: String
+        get() = actual.url
+
+    /**
+     * Restores a single tab from this collection and returns a matching returns a matching [SessionManager.Snapshot].
```
Comment (Amejia481) [inline components/feature/tab-collections/src/main/java/mozilla/components/feature/tab/collections/adapter/TabAdapter.kt]: nit: **returns a matching** returns a matching

File: components/feature/tab-collections/src/androidTest/java/mozilla/components/feature/tab/collections/TabCollectionStorageTest.kt (hunk 1 above)
Comment (pocmo) [reply components/feature/tab-collections/src/androidTest/java/mozilla/components/feature/tab/collections/TabCollectionStorageTest.kt]: It seemed fine when testing in the sample browser. I think that may be caused by the test setup where I tried to get a list from the datasource for testing.

File: components/feature/tab-collections/src/main/java/mozilla/components/feature/tab/collections/Tab.kt (hunk 2 above)
Comment (pocmo) [reply components/feature/tab-collections/src/main/java/mozilla/components/feature/tab/collections/Tab.kt]: Updated!

File: components/feature/tab-collections/src/main/java/mozilla/components/feature/tab/collections/adapter/TabAdapter.kt (hunk 3 above)
Comment (pocmo) [reply components/feature/tab-collections/src/main/java/mozilla/components/feature/tab/collections/adapter/TabAdapter.kt]: Updated!

File: components/feature/tab-collections/src/main/java/mozilla/components/feature/tab/collections/adapter/TabAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+import android.content.Context
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.browser.session.ext.readSnapshotItem
+import mozilla.components.concept.engine.Engine
+import mozilla.components.feature.tab.collections.Tab
+import mozilla.components.feature.tab.collections.db.TabEntity
+
+internal class TabAdapter(
+    val actual: TabEntity
```
Comment (Amejia481) [inline components/feature/tab-collections/src/main/java/mozilla/components/feature/tab/collections/adapter/TabAdapter.kt]: Could renaming  `actual` to `tabEntity`  convey is purpose better? In the diff the first time that I saw `actual` been use from another class I thought that you  multiple instance and `actual` was the `actual` one.

File: components/feature/tab-collections/src/androidTest/java/mozilla/components/feature/tab/collections/TabCollectionStorageTest.kt (hunk 1 above)
Comment (Amejia481) [reply components/feature/tab-collections/src/androidTest/java/mozilla/components/feature/tab/collections/TabCollectionStorageTest.kt]: 👍

File: components/feature/tab-collections/src/main/java/mozilla/components/feature/tab/collections/adapter/TabAdapter.kt (hunk 4 above)
Comment (pocmo) [reply components/feature/tab-collections/src/main/java/mozilla/components/feature/tab/collections/adapter/TabAdapter.kt]: I renamed the property in both adapters to "entity". :)

Comment (Amejia481) [reply components/feature/tab-collections/src/main/java/mozilla/components/feature/tab/collections/adapter/TabAdapter.kt]: 👍

**Pull request merged**